### DISCUSSION TITLE
Dedicated semantic token for method declaration with property decorator

### DISCUSSION THREAD

**Discussion by dmrowiec-pl** [microsoft/pylance-release#2640] [Enhancement]
Currently, when method inside class is decorated with `@property`, it's semantic token type is set as 'property', and modifier as 'declaration'. That is consistent with the actual state, however since in code it's written as a method, there is no possibility to override the color assigned to this particular scenario (modifying color set for 'property declaration' would just override declaration of every single variable set on a class instance). As a result, you get output like on the image below:

![semantic_color](https://user-images.githubusercontent.com/51089855/109397193-d80eac80-7935-11eb-9a21-88619cd1931b.jpg)

You can see that it looks kind of weird. Anyway, it would be nice to add another modifier to the chain, like: 'property'->'declaration'->'decorator', so that it could be modified and set by the end-user in settings. Referring to the image above, that would allow to set the color to yellow for 'test_property' in lines 11 and 15, in order to have more consistency.

**Comment by ian-h-chamberlain** [comment]
This seems somewhat related, but perhaps deserves its own issue if not: class constants seem to be marked as `property` as well, which doesn't allow any color customization on them either:

![Screen Shot 2022-01-06 at 5 16 02 PM](https://user-images.githubusercontent.com/11131775/148460078-9c191e60-db93-4aaf-9a41-301aad3e36e6.png)

One exception is enum members, which seem to have the special token type `enumMember`, which is highlighted like `variable` by default in my theme but can be customized:

![Screen Shot 2022-01-06 at 5 17 59 PM](https://user-images.githubusercontent.com/11131775/148460196-226a03a6-456a-4ee0-88d7-520f28e8ffe1.png)

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
